File: legacy/src/main/java/com/redbox/global/util/RandomCodeGenerator.java
package com.redbox.global.util;

import java.security.SecureRandom;

public class RandomCodeGenerator {
    private static final String CHARACTERS = "abcdefghijklmnopqrstuvwxyz0123456789";
    private static final SecureRandom SECURE_RANDOM = new SecureRandom();
    private static final int CODE_LENGTH = 8;

    public static String generateRandomCode() {
        StringBuilder randomCode = new StringBuilder();

        for (int i = 0; i < CODE_LENGTH; i++) {
            int index = SECURE_RANDOM.nextInt(CHARACTERS.length());
            randomCode.append(CHARACTERS.charAt(index));
        }

        return randomCode.toString();
    }
}

File: migration/src/main/kotlin/com/redbox/global/util/RandomCodeGenerator.kt
package com.redbox.global.util

import java.security.SecureRandom

private const val CHARACTERS: String = "abcdefghijklmnopqrstuvwxyz0123456789"
private val SECURE_RANDOM: SecureRandom = SecureRandom()
private const val CODE_LENGTH: Int = 8

class RandomCodeGenerator {

    companion object {
        fun generateRandomCode(): String {
            val randomCode: StringBuilder = StringBuilder()

            for (i in 0 until CODE_LENGTH) {
                val index = SECURE_RANDOM.nextInt(CHARACTERS.length)
                randomCode.append(CHARACTERS[index])
            }

            return randomCode.toString()
        }
    }
}